coretex/coretex/sample/sequence_sample/local_sequence_sample.py

```diff
@@ -68,3 +114,55 @@ def unzip(self, ignoreCache: bool = False) -> None:
             file_utils.gzipDecompress(compressedSequencePath, decompressedSequencePath)
         except FileNotFoundError:
             pass
+
+    def __unzipPairedEnd(self, ignoreCache: bool = False) -> None:
+        super().unzip(ignoreCache)
+
+        try:
+            compressedForwardPath = getForwardSequenceFile(Path(self.path), [".gz"])
+            compressedReversePath = getReverseSequenceFile(Path(self.path), [".gz"])
+
+            for compressedSequencePath in [compressedForwardPath, compressedReversePath]:
+                decompressedSequencePath = compressedSequencePath.parent / compressedSequencePath.stem
+
+                if decompressedSequencePath.exists() and not ignoreCache:
+                    return
+
+                file_utils.gzipDecompress(compressedSequencePath, decompressedSequencePath)
+        except FileNotFoundError:
+            pass
+
+    def unzip(self, ignoreCache: bool = False) -> None:
+        self.__unzipPairedEnd(ignoreCache) if self.isPairedEnd() else self.__unzipSingleEnd(ignoreCache)
+
+    def isPairedEnd(self) -> bool:
+        """
+            This function returns True if the sample holds paired-end reads and
+            False if it holds single end. Files for paired-end reads must contain
+            "_R1_" and "_R2_" in their names, otherwise an exception will be raised
+            If the sample contains gzip compressed sequences, you will have to call
+            Sample.unzip method first otherwise calling Sample.isPairedEnd will
+            raise an exception
+
+            Raises
+            ------
+            FileNotFoundError -> if no files meeting the requirements for either single-end
+            or paired-end sequencing reads
+        """
+
+        extensions = self.supportedExtensions()
+        for extension in extensions:
+            listOfPaths = list(self.path.glob(f"*{extension}"))
+
+            if len(listOfPaths) == 0:
+                continue
+
+            if len(listOfPaths) == 1:
+                return False
+
+            if len(listOfPaths) == 2:
+                for path in listOfPaths:
+                    if "_R1_" not in path.name and "_R2_" not in path.name:
```

Comment:
Instead of looping over it we should manually check for "_R1_" and "_R2_" patterns in filenames. This logic will return True for a sample which contains 2 "_R1_" files.